store/multiversion/mvkv.go

```diff
@@ -316,7 +316,28 @@ func (v *VersionIndexedStore) VersionExists(version int64) bool {
 }
 
 func (v *VersionIndexedStore) DeleteAll(start, end []byte) error {
-	return v.parent.DeleteAll(start, end)
+	for _, k := range v.GetAllKeyStrsInRange(start, end) {
+		v.Delete([]byte(k))
+	}
+	return nil
+}
+
+func (v *VersionIndexedStore) GetAllKeyStrsInRange(start, end []byte) (res []string) {
+	keyStrs := map[string]struct{}{}
+	for _, pk := range v.parent.GetAllKeyStrsInRange(start, end) {
+		keyStrs[pk] = struct{}{}
+	}
```

Comment:
we need to also consider the multiVersionStore when getting all the keys in range, since earlier txs potentially may have written within the range or deleted keys within the range, both of which should affect the result of this function. As such, we would also need to update readsets here, right? we should add all of these keys read both from parent OR from multiversion store to the readset so that the DeleteAll is properly backed by an updated readset representing the accesses of those keys / values

Reply:
actually, we should do something like iterateset in this case, because the keys that are accessed within the range is important here

Reply:
I think the easiest solution in this case (perhaps not the most optimal but should be correct) would be to create an Iterator within deleteAll, iterate over the whole iterator collecting keys, and THEN call delete on each one. Then you get the iteration safety out of the box